tapgarden/planter.go

```diff
@@ -595,12 +581,33 @@ func (c *ChainPlanter) gardener() {
 				log.Infof("Finalizing batch %x",
 					batchKey.SerializeCompressed())
 
-				// A force tick must be sent in a separate
-				// goroutine to prevent deadlock.
-				go func() {
-					c.cfg.BatchTicker.Force <- time.Time{}
-				}()
-				req.Resolve(batchKey)
+				caretaker, err := c.finalizeBatch()
```

Comment:
Not sure I understand why this logic is added here and not in the existing case in the surrounding `case` statement here: https://github.com/lightninglabs/taproot-assets/blob/c8bed246d2f3db1fd9c12cc85d0dcada18b65d59/tapgarden/planter.go#L470

One issue seems to be returning state from that case, but as-is finalizing the batch via the batch ticker will also not handle possible broadcast errors on the new `BroadcastErrChan`.

Reply:
Because the `c.cfg.BatchTicker.Ticks()` path is only called on startup, there is no "caller" to return errors to. So I don't think we need to explicitly wait for an error to be returned, as it will only be logged anyway. See my comment below about potentially removing the ticker path completely? Because IMO a batch shouldn't suddenly be completed just because the daemon is restarted.